# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/arcticfox/hotfix-helperthreads-join.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- Arctic-Fox-46.2/js/src/vm/HelperThreads.cpp.vanilla	2025-09-05 17:08:36.414708897 +0200
+++ Arctic-Fox-46.2/js/src/vm/HelperThreads.cpp	2025-09-05 17:21:41.058517716 +0200
@@ -1288,7 +1288,10 @@
             HelperThreadState().notifyAll(GlobalHelperThreadState::PRODUCER, lock);
         }
 
-        thread->join();
+        if (thread->joinable())
+            /* If thread has not exited yet, wait for it */
+            thread->join();
+
         thread.reset();
     }
 
